app/main.py

```diff
@@ -88,6 +91,29 @@ def _get_client(self):
                                              self.client_id, self.client_secret)
         return KeyVaultClient(credentials)
 
+    def _get_tenant_id(self):
+        vault_base_url = os.getenv('VAULT_BASE_URL')
+        _logger.info('Detecting tenant id for %s', vault_base_url)
+        # Some key to trigger a 401
+        URL = '{}/keys/somekeyname/1?api-version=2018-02-14'.format(vault_base_url)
+        _logger.info('Sending challenge request to %s', URL)
+        response = requests.get(url = URL)
+        if response.status_code == 401:
```

Comment:
you should probably handle other status codes as well, or at least write them to log

Reply:
done. Logged other status codes. Also in case the tenant could not be retrieved, will default to the tenant id from config.